tpke/src/api.rs

```diff
@@ -4,144 +4,108 @@
 
 // TODO: Refactor this module to deduplicate shared code from tpke-wasm and tpke-wasm.
 
-use std::convert::TryInto;
-
-use ark_ec::{AffineCurve, ProjectiveCurve};
-use ark_ff::{BigInteger256, ToBytes};
-
-// Fixing some of the types here on our target engine
-// TODO: Consider fixing on crate::api level instead of bindings level
-type E = ark_bls12_381::Bls12_381;
-type TpkePublicKey = ark_bls12_381::G1Affine;
-type TpkePrivateKey = ark_bls12_381::G2Affine;
-type TpkeCiphertext = crate::Ciphertext<E>;
-type TpkeDecryptionShare = crate::DecryptionShareFast<E>;
-type TpkePublicDecryptionContext = crate::PublicDecryptionContextFast<E>;
-type TpkeSharedSecret =
-    <ark_bls12_381::Bls12_381 as ark_ec::PairingEngine>::Fqk;
-
-#[derive(Clone, Debug)]
-pub struct PrivateDecryptionContext {
-    pub b_inv: ark_bls12_381::Fr,
-    pub decrypter_index: usize,
+use ark_serialize::*;
+
+pub type E = ark_bls12_381::Bls12_381;
+pub type TpkeDkgPublicKey = ark_bls12_381::G1Affine;
+pub type TpkeG1Prepared = <E as ark_ec::PairingEngine>::G1Prepared;
+pub type TpkeG1Affine = <E as ark_ec::PairingEngine>::G1Affine;
+pub type TpkePrivateKey = ark_bls12_381::G2Affine;
+pub type TpkeUnblindingKey = ark_bls12_381::Fr;
+pub type TpkeDomainPoint = ark_bls12_381::Fr;
+pub type TpkeCiphertext = crate::Ciphertext<E>;
+pub type TpkeDecryptionShareSimplePrecomputed =
+    crate::DecryptionShareSimplePrecomputed<E>;
+pub type TpkeDecryptionShareSimple = crate::DecryptionShareSimple<E>;
+pub type TpkePublicDecryptionContext = crate::PublicDecryptionContextSimple<E>;
+pub type TpkeSharedSecret = <E as ark_ec::PairingEngine>::Fqk;
+pub type TpkeResult<T> = crate::Result<T>;
+pub type TpkePrivateDecryptionContext =
+    crate::PrivateDecryptionContextSimple<E>;
+
+pub fn encrypt(
+    message: &[u8],
+    aad: &[u8],
+    public_key: &TpkeDkgPublicKey,
+) -> Ciphertext {
+    // TODO: Should rng be a parameter?
+    let rng = &mut rand::thread_rng();
+    Ciphertext(crate::encrypt(message, aad, public_key, rng))
 }
 
-impl PrivateDecryptionContext {
-    const B_INV_LEN: usize = 32;
-    const DECRYPTER_INDEX_LEN: usize = 8;
+pub fn decrypt_with_shared_secret(
+    ciphertext: &Ciphertext,
+    aad: &[u8],
+    shared_secret: &TpkeSharedSecret,
+    g_inv: &TpkeG1Prepared,
+) -> TpkeResult<Vec<u8>> {
+    crate::decrypt_with_shared_secret(&ciphertext.0, aad, shared_secret, g_inv)
+}
 
-    pub fn new(b_inv: &ark_bls12_381::Fr, decrypter_index: usize) -> Self {
-        Self {
-            b_inv: *b_inv,
-            decrypter_index,
-        }
-    }
+pub fn decrypt_symmetric(
```

Comment:
why do we expose both `decrypt_with_shared_secret` and `decrypt_symmetric`?

Reply:
I'm using `decrypt_symmetric` as a sanity check in e2e testing, we will probably not ship this method in the final version of the API